src/components/assets/icons/SSIEntityIcon/index.tsx

```diff
@@ -1,29 +1,41 @@
 import React, { FC } from 'react'
-import { ColorValue, View } from 'react-native'
-import Svg, { Circle, Path } from 'react-native-svg'
+import { ColorValue } from 'react-native'
+import { connect } from 'react-redux'
 
 import { entities, fonts } from '../../../../styles/colors'
+import {
+  SSIEntityIconStyled as Avatar,
+  SSITextH2LightStyled as AvatarText
+} from '../../../../styles/components'
+import { RootState } from '../../../../types'
+import { getDefaultUserNameInitials } from '../../../../utils/UserUtils'
 
 export interface IProps {
   size?: number
-  color?: ColorValue
-  backgroundColor?: ColorValue
+  fontColor?: ColorValue
+  bgColor?: ColorValue
+  initialsOfUserName: string
 }
 
 const SSIEntityIcon: FC<IProps> = (props: IProps): JSX.Element => {
-  const { size = 34, color = fonts.light, backgroundColor = entities['100'] } = props
+  const { size = 34, fontColor = fonts.light, bgColor = entities['100'], initialsOfUserName = '?' } = props
 
   return (
-    <View style={{ width: size, aspectRatio: 1 }}>
-      <Svg width="100%" height="100%" viewBox="0 0 34 34" fill="none">
-        <Circle cx="17" cy="17" r="17" fill={backgroundColor} />
-        <Path
-          d="M16.7344 23.0781H15.0156L9.03906 13.6875V23H7.57031V11.7656H9.28906L15.2656 21.1562V11.7656H16.7344V23.0781ZM27.1719 22.6797C26.6875 22.8307 26.1745 22.9531 25.6328 23.0469C25.0964 23.1406 24.5052 23.1875 23.8594 23.1875C22.2656 23.1875 21.0625 22.7109 20.25 21.7578C19.4375 20.8047 19.0312 19.401 19.0312 17.5469C19.0312 15.5781 19.4375 14.0938 20.25 13.0938C21.0625 12.0938 22.2656 11.5938 23.8594 11.5938C24.4635 11.5938 25.0052 11.6458 25.4844 11.75C25.9688 11.8542 26.4349 11.9896 26.8828 12.1562L26.4766 13.3125C26.263 13.2344 26.0521 13.1641 25.8438 13.1016C25.6354 13.0391 25.4245 12.987 25.2109 12.9453C25.0026 12.9036 24.7865 12.8724 24.5625 12.8516C24.3385 12.8307 24.1042 12.8203 23.8594 12.8203C23.2812 12.8203 22.7839 12.9245 22.3672 13.1328C21.9505 13.3359 21.6068 13.6354 21.3359 14.0312C21.0651 14.4219 20.8646 14.901 20.7344 15.4688C20.6042 16.0312 20.5391 16.6719 20.5391 17.3906C20.5391 18.1302 20.6042 18.7839 20.7344 19.3516C20.8646 19.9193 21.0651 20.3984 21.3359 20.7891C21.6068 21.1745 21.9505 21.4661 22.3672 21.6641C22.7839 21.862 23.2812 21.9609 23.8594 21.9609C24.3906 21.9609 24.8958 21.9167 25.375 21.8281C25.8542 21.7396 26.3307 21.6224 26.8047 21.4766L27.1719 22.6797Z"
-          fill={color}
-        />
-      </Svg>
-    </View>
+    <Avatar
+      style={{
+        backgroundColor: bgColor,
+        width: size
+      }}
+    >
+      <AvatarText style={{ color: fontColor }}>{initialsOfUserName}</AvatarText>
+    </Avatar>
   )
 }
 
-export default SSIEntityIcon
+const mapStateToProps = (state: RootState) => {
+  return {
+    initialsOfUserName: getDefaultUserNameInitials(state.user.activeUser?.firstName, state.user.activeUser?.lastName)
+  }
+}
+
+export default connect(mapStateToProps, null)(SSIEntityIcon)
```

Comment:
I did not fully understand where do we want to move it ? I mean if I pass it as prop from Header then header should also receive it from header's parent. Which will trigger a long chain. I mean we will be passing initials from like 5th - 10th parent of the avatar ... perhaps we can discuss further. As this part is not clear to me.

Reply:
ye, i guess that using the redux store is justified here to avoid passing it as a route prop everytime.

so lets keep the connect. look at examples where we use connect, that component is an extension of a PureComponent. Use that structure here as well.

Reply:
I am confirming that we did not need to convert the functional component to the pure component, right? we just needed to follow the connect pattern used in pure component?

Reply:
Connect has been used differently now.